Mathlib/Analysis/Normed/Ring/Basic.lean

```diff
@@ -263,6 +304,38 @@ instance (priority := 75) NonUnitalSubalgebraClass.nonUnitalNormedRing {S 𝕜 E
   { nonUnitalSeminormedRing s with
     eq_of_dist_eq_zero := eq_of_dist_eq_zero }
 
+instance ULift.nonUnitalNonAssocSeminormedRing : NonUnitalNonAssocSeminormedRing (ULift α) :=
+  { ULift.seminormedAddCommGroup, ULift.nonUnitalNonAssocRing with
+    norm_mul_le x y := norm_mul_le x.down y.down }
+
+/-- Non-unital, non-associative seminormed ring structure on the product of two non-unital,
+non-associative seminormed rings, using the sup norm. -/
+instance Prod.nonUnitalNonAssocSeminormedRing [NonUnitalNonAssocSeminormedRing β] :
+    NonUnitalNonAssocSeminormedRing (α × β) :=
+  { seminormedAddCommGroup, instNonUnitalNonAssocRing with
+    norm_mul_le x y := calc
+      ‖x * y‖ = ‖(x.1 * y.1, x.2 * y.2)‖ := rfl
+      _ = max ‖x.1 * y.1‖ ‖x.2 * y.2‖ := rfl
+      _ ≤ max (‖x.1‖ * ‖y.1‖) (‖x.2‖ * ‖y.2‖) :=
+        (max_le_max (norm_mul_le x.1 y.1) (norm_mul_le x.2 y.2))
+      _ = max (‖x.1‖ * ‖y.1‖) (‖y.2‖ * ‖x.2‖) := by simp [mul_comm]
+      _ ≤ max ‖x.1‖ ‖x.2‖ * max ‖y.2‖ ‖y.1‖ := by
+        apply max_mul_mul_le_max_mul_max <;> simp [norm_nonneg]
+      _ = max ‖x.1‖ ‖x.2‖ * max ‖y.1‖ ‖y.2‖ := by simp [max_comm]
+      _ = ‖x‖ * ‖y‖ := rfl }
```

Comment:
It's fine to copy this proof here, but then delete the one below for the associative version. No reason to duplicate.